src/main/java/de/unijena/cheminf/fragment/fingerprint/FragmentFingerprinter.java

```diff
@@ -153,23 +155,142 @@ public FragmentFingerprinter(List<String> aFragmentList) throws NullPointerExcep
+            if (aSmilesToPositionMap.containsKey(tmpSmiles)) {
+                aPreInitFloatArray[aSmilesToPositionMap.get(tmpSmiles)] = 1.0f;
+            }
+        }
+        return aPreInitFloatArray;
+    }
+    //
+    /**
+     * Public method for generating a float count fingerprint in form of a float[] array.
+     * This fingerprint projects the frequency (-> count) of occurrence of the pre-defined fingerprint fragments within
+     * the given SMILES.
+     * <p>
+     * For example: The pre-defined fragment fingerprint contains the SMILES for methane as "C", for an ether substructure
+     * as "*O*", and for a hydroxide group as "[H]OC". The given SMILES list contains only three "C" fragments. Then,
+     * the generated fingerprint would look like the following: [3.0f, 0.0f, 0.0f]
+     * </p>
+     *
+     * @param aFragmentsUniqueSmilesList of fragment SMILES to be compared to the pre-defined fingerprint
+     * @param aSmilesToPositionMap       mapping pre-defined fingerprint fragments to their position inside the fingerprint
+     * @param aPreInitFloatArray         pre-initialized float[] in which the fingerprint is to be stored
+     * @return float[] array filled with float values representing the generated fingerprint
+     */
+    public float[] getFloatCountFingerprint(
+            List<String> aFragmentsUniqueSmilesList,
+            Map<String, Integer> aSmilesToPositionMap,
+            float[] aPreInitFloatArray) {
+        for (String tmpSmiles : aFragmentsUniqueSmilesList) {
+            if (aSmilesToPositionMap.containsKey(tmpSmiles)) {
+                aPreInitFloatArray[aSmilesToPositionMap.get(tmpSmiles)]++;
+            }
+        }
+        return aPreInitFloatArray;
+    }
+    //
+    /**
+     * Public method to get a float[][] matrix containing the fingerprints for the given SMILES in regards to the
+     * pre-defined fragment fingerprint.
+     * It shall be noted that each List in the given List (List of Lists) represents ONE molecule's fragments.
+     * Therefor, the whole list represents the entirety of fragments.
+     * Further, a setting whether to use "bit set" or "count/frequency of fragment" is available.
+     *
+     * @param aFragmentsUniqueSmilesListsArrayList with Lists of SMILES to compare with the defined fingerprint
+     * @param aFloatDataMatrix to be filled with the float fingerprints of the given fragments
+     * @param anUseBitArrayStatement setting whether "bit set" or "count/frequency" should be used for the matrix
+     * @return float[][] matrix, filled with float values from the generated bit set or count/frequency set
+     */
+    public float[][] getFragmentsComponentsFloatMatrix(
+            List<List<String>> aFragmentsUniqueSmilesListsArrayList,
+            float[][] aFloatDataMatrix,
+            boolean anUseBitArrayStatement
+    ) {
+        //float[row count][column count]
+        for (int i = 0; i < aFloatDataMatrix.length; i++) {
```

Comment:
for this to work, you should add a parameter check that the list of lists and the matrix row count are the same.